.github/workflows/main.yml

```diff
@@ -1,68 +1,67 @@
-  # Triggers the workflow on pull request events and merges/pushes to main
   push:
     branches:
-      - 'octopus'
+      - octopus
   pull_request:
     types: [opened, synchronize, reopened]
   schedule:
     # Daily 5am australian/brisbane time
     - cron: '0 19 * * *'
   workflow_dispatch:
-
+    # Allows you to run this workflow manually from the Actions tab
+ 
 env:
-  OCTOVERSION_CurrentBranch: ${{ github.head_ref || github.ref }} # For pull_request events we override the /refs/pull/xx/merge branch to the PR's head branch
+  OCTOVERSION_CurrentBranch: ${{ github.head_ref || github.ref }}
+  OCTOPUS_SPACE: "Core Platform"
 
 jobs:
   build:
     runs-on: ubuntu-latest
     permissions:
       id-token: write # Required to obtain the ID token from GitHub Actions
-      contents: read # Required to check out code
+      contents: write # Read Required to check out code, Write to create Git Tags
       checks: write # Required for test-reporter
-    env:
-      OCTOPUS_SPACE: "Core Platform"
     steps:
-      - uses: actions/checkout@v4
+      # Must clone the entire history for OctoVersion to work
+      - name: Checkout
+        uses: actions/checkout@v4
         with:
-          fetch-depth: 0 # all
+          fetch-depth: 0 
 
       - name: Setup .NET
-        uses: actions/setup-dotnet@v3
+        uses: actions/setup-dotnet@v4
         with:
           dotnet-version: 6.0.x
-      
-      # Adjustment is done prior to Nuke build as OCTOVERSION information is included in the result package.
-      - name: Append OCTOVERSION_CurrentBranch with -nightly (for scheduled)
+
+      - name: Append OCTOVERSION_CurrentBranch with -nightly-<timestamp> (for scheduled)
         if: github.event_name == 'schedule'
-        run: echo "OCTOVERSION_CurrentBranch=${{ env.OCTOVERSION_CurrentBranch }}-nightly-$(date +'%Y%m%d%H%M%S')" >> $GITHUB_ENV
+        run: |
+          echo "OCTOVERSION_CurrentBranch=${{ env.OCTOVERSION_CurrentBranch }}-nightly-$(date +'%Y%m%d%H%M%S')" >> $GITHUB_ENV
 
       - name: Nuke Build 🏗
         id: build
-        shell: bash                
-        env:          
-          OCTOVERSION_Patch: ${{ github.run_number }} # Pass patch number to Nuke OctoVersion
         run: ./build.sh
 
-      - name: Linux unit test report
+      - name: Publish Test Results
         uses: dorny/test-reporter@v1
         if: success() || failure()    # run this step even if previous step failed
         with:
-          name: Linux unit test results
+          name: Test Results
           path: ./TestResults/*.trx
           reporter: dotnet-trx
           fail-on-error: true
-        
-      - name: Tag release (when not pre-release) 🏷️
-        id: github-tag        
-        if: ${{ github.event_name != 'schedule' && !contains( steps.build.outputs.octoversion_fullsemver, '-' ) }}
+
+      - name: Git Tag (when not pre-release) 🏷️
+        id: github-tag
+        if: ${{ !contains( steps.build.outputs.octoversion_fullsemver, '-' ) }}
```

Comment:
The choice to not tag nightlies (`github.event_name != 'schedule'`) seems intentional, and I can imagine why as it could lead to many redundant tags on the one commit. What's your reasoning for removing it?

Reply:
Removed because it's redundant (and to sync with the other repos, where I also removed it).

Further up, the "Append nightly-timestamp" step runs if `github.event_name == 'schedule'` which means that when we arrive here, scheduled builds will always have a hyphen in the build number on account of that anyway